### PR TITLE
Partial Fix for Issue 16073 (part 2)

### PR DESCRIPTION
**Pull request by JackStouffer**
That does it for std.algorithm

### CODE REVIEW

File: std/algorithm/sorting.d [hunk 1]
```diff
@@ -387,8 +387,8 @@ See_Also:
 */
 Range partition(alias predicate,
         SwapStrategy ss = SwapStrategy.unstable, Range)(Range r)
-    if ((ss == SwapStrategy.stable && isRandomAccessRange!(Range))
-            || (ss != SwapStrategy.stable && isForwardRange!(Range)))
+    if ((ss == SwapStrategy.stable && isRandomAccessRange!(Range) && hasLength!Range && hasSlicing!Range)
+            || isInputRange!(Range))
```
Comment (wilzbach) [inline std/algorithm/sorting.d]: for a  _stable_ swap strategy RandomAccess is needed, we shouldn't relax this concern ;-)

**Comment by wilzbach** [comment]
small nitpick, and apart from the usual regret that we should fix `$` on a language basis (it makes the code is bit more "D-ish"), LGTM!

**Comment by JackStouffer** [comment]
Fixed

**Comment by schveiguy** [comment]
`topN` has `r[0 .. $]` in its docs. Fix and then I think this LGTM.

https://github.com/JackStouffer/phobos/blob/65c8d710b927a22c3a678aa2ce8515d4dbd815ef/std/algorithm/sorting.d#L2325

**Comment by schveiguy** [comment]
Yay github for sending emails for new commits! 👍

**Comment by JackStouffer** [comment]
Fixed

**Comment by schveiguy** [comment]
Auto-merge toggled on

**Comment by JackStouffer** [comment]
Thanks!

**Pull request merged**